Split tests into separate files

kuzmacska opened the pull request.

akaRem commented on nexus/resource_repository_dockerhostedwithoutports_test.go:
```diff
@@ -0,0 +1,25 @@
+	"fmt"
+	"strconv"
+)
+
```
You forgot to include actual test.

on L162 : `func TestAccRepositoryDockerHostedWithoutPorts(t *testing.T) { ... `

akaRem commented on nexus/resource_repository_apthosted_test.go:
```diff
@@ -0,0 +1,65 @@
+	})
+}
+
+func testAccRepositoryAptHosted(name string, aptDistribution string, aptSigningKEypair string, aptSigningPassphrase string, cleanupPolicyNames []string) string {
```
These helper functions may be renamed to smth like `createTerraformStatementForResourceAptHosted(...)` or may be even shorter `createTfStmtForResourceAptHosted()` to reflect their purposes better (they are not private tests, they are resource statements generators).

akaRem commented on nexus/resource_repository_apthosted_test.go:
```diff
@@ -0,0 +1,65 @@
+	format = "apt"
+	type   = "hosted"
+
+    apt {
```
Could you please reformat? There are mixed spaces and tabs which produce not very readable text (switch all to tabs)

akaRem commented on nexus/resource_repository_dockerproxy_test.go:
```diff
@@ -0,0 +1,67 @@
+		v1enabled        = false
+	}
+
+    docker_proxy {
```
same issue with formatting here

akaRem commented on nexus/resource_repository_apthosted_test.go:
```diff
@@ -0,0 +1,65 @@
+				ImportStateId:     repoName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				// apt_signing not returned by API
```
While you are moving things around and you are already have full overview of tests, could you please also add some notes about the current state of tests and things which are missing here:
- which fields are not being checked?
- which configurations are not being tested?

ex:
```go
// TODO: verify apt_signing configuration, apt_signing not returned by API currently
ImportStateVerifyIgnore: []string{"apt_signing"},
// TODO: add check for storage
// TODO: add check for apt.distribution
// TODO: add tests for readonly repository
```
Then later we will not forget about things to be done.

akaRem reviewed (changes requested):
Please double check the list of tests before and after.

And please also rewrite your commits in format compliant with history of repo https://github.com/datadrivers/terraform-provider-nexus/commits/master

Commit:
Refactor nexus_registry tests

Currently all the tests are put into one file.
It's hard to read and browse tests related to
different types and formats of repository provider.
There are also some missing checks.

This change addresses this problem and
- split tests
- add some todos
- rename helper functions
- fix some indents
